File: leetcode/2303-CalculateAmountPaidInTaxes/2303.c
double calculateTax(int** brackets, int bracketsSize, int* bracketsColSize, int income){
    int paid = 0; 
    int prev_upper = 0; 
    for(int i = 0; i < bracketsSize; ++i)
    {
        if(brackets[i][0] > income)
        {
            paid += (income - prev_upper) * brackets[i][1]; 
            break; 
        }
        paid += (brackets[i][0] - prev_upper) * brackets[i][1]; 
        prev_upper = brackets[i][0]; 
    }
    return ( (double) paid) / 100.0; 
}

File: leetcode/2303-CalculateAmountPaidInTaxes/2303.py
from typing import List 

class Solution:
    def calculateTax(self, brackets: List[List[int]], income: int) -> float:
        if income == 0: 
            return 0.0 
        paid: int = 0 
        prev_upper: int = 0 
        for upper, percent in brackets: 
            if upper > income: 
                paid += ( income - prev_upper ) * percent 
                break  
            paid += ( upper - prev_upper ) * percent
            prev_upper = upper 
        return float(paid) / 100.0